R/grob-dotstack.r

```diff
@@ -13,7 +13,7 @@ dotstackGrob <- function(
         y <- unit(y, default.units)
     if (!is.unit(dotdia))
         dotdia <- unit(dotdia, default.units)
-    if (attr(dotdia,"unit") != "npc")
+    if (!grepl('npc', as.character(dotdia))) # Only cross-version reliable way to check the unit of a unit object
```

Comment:
Maybe wrap this in a function and make the version dependency explicit by executing different code paths for different version? This would enable us to update to a more elegant check in the future once we stop supporting the old R versions that require the workaround.

Reply:
The old one was also a hack that could fail (but didn’t for the specific use case). This check is cross-version so I can’t see the benefit of having separate approaches..?

Reply:
I see. In any case, just a suggestion: I usually prefer to have such tests encapsulated in a function. Makes the code more readable and easier to maintain.

Reply:
I can see the point of encapsulation - will do